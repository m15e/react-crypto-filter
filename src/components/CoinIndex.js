import '../App.css';
import CoinList from '../containers/CoinList';

function CoinIndex() {
  return (
    <div className="coin-index">
      <CoinList />
    </div>
  );
}

export default CoinIndex;
