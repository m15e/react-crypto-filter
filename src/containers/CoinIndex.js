import '../App.css';
import CoinList from './CoinList';

function CoinIndex() {
  return (
    <div className="coin-index">
      <CoinList />
    </div>
  );
}

export default CoinIndex;
