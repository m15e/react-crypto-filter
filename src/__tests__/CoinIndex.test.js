import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom';
import CoinIndex from '../components/CoinIndex';

const mockStore = configureStore();

const store = mockStore({
  coins: {
    items: [
      {
        id: 'bitcoin',
        symbol: 'btc',
        name: 'Bitcoin',
        image:
          'https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579',
        current_price: 44497,
        market_cap: 830911434449,
        ath: 48025,
      },
    ],
  },
  filter: {
    filter: 'Show All',
    search: '',
  },
});

it('CoinIndex matches Snapshot', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Provider store={store}>
          <CoinIndex />
        </Provider>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
