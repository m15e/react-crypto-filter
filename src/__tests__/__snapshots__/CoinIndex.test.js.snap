// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinIndex matches Snapshot 1`] = `
<div
  className="coin-index"
>
  <div
    className="coinList container"
  >
    <div
      className="container filter-container"
    >
      <div
        className="columns"
      >
        <div
          className="column"
        >
          <p>
            Search:
          </p>
        </div>
        <div
          className="column"
        >
          <input
            name="search"
            onChange={[Function]}
            placeholder="Enter Coin Name"
            type="text"
          />
        </div>
      </div>
      <div
        className="columns"
      >
        <div
          className="column"
        >
          <p>
            Filter Top 10 by:
          </p>
        </div>
        <div
          className="column"
        >
          <select
            name="coinFilter"
            onChange={[Function]}
          >
            <option
              value="Show All"
            >
              Show All
            </option>
            <option
              value="Market Cap"
            >
              Market Cap
            </option>
            <option
              value="Dollar Value"
            >
              Dollar Value
            </option>
            <option
              value="All Time High"
            >
              All Time High
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      className="coin-container"
    >
      <div
        className="coin tile is-child box is-4"
        id="bitcoin"
      >
        <figure
          className="image is-64x64 coin-logo"
        >
          <img
            alt="Bitcoin"
            className="is-rounded"
            src="https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579"
          />
        </figure>
        <h3
          className="title is-5"
        >
          Bitcoin
        </h3>
        <p>
          <span>
            Market Capitalization
          </span>
          <br />
          $830,911,434,449.00
        </p>
        <p>
          <span>
            Current Price
          </span>
          <br />
          $44,497.00
        </p>
        <p>
          <span>
            All Time High
          </span>
          <br />
          $48,025.00
        </p>
        <a
          className="view-coin button is-ghost is-small"
          href="/coin/bitcoin"
          onClick={[Function]}
        >
          View Coin
        </a>
      </div>
    </div>
  </div>
</div>
`;
